pkg/dsync/retry.go

```diff
@@ -105,13 +104,10 @@ func newRetryTimerWithJitter(ctx context.Context, unit time.Duration, cap time.D
 				// Stop the routine.
 				return
 			}
-			timer = time.NewTimer(exponentialBackoffWait(nextBackoff))
-			// wait till next backoff time or till doneCh gets a message.
+			// wait till next backoff time or till context is canceled.
 			select {
-			case <-timer.C:
+			case <-time.After(exponentialBackoffWait(nextBackoff)):
```

Comment:
Reusing a timer isannoying, but doable.

Reply:
the point is once the scope is lost it should recover @klauspost this is the same as using `time.NewTimer().C in data-usage.go` @klauspost our interest is simply to sleep here and cancel if context is finished. 

The previous code issue was that `timer` var was outside of for loop allowing the scope to be valid and never be garbage collected.

Reply:
It will spawn a goroutine that will keep it alive. You need to Stop and reset the timer instead if you loop.
This change does nothing AFAICT.

You need this: https://golang.org/pkg/time/#Timer.Reset

Reply:
Re-implemented it differently. PTAL @klauspost